src/S3.ts

```diff
@@ -16,43 +17,51 @@ class S3 {
   }
 
   public async initialise() {
-    const s3UrlBase: any = new S3File.Endpoint(this.url)
-    this.s3Handler = new S3File.S3({
-      endpoint: s3UrlBase,
-      accessKeyId: this.params.keyId,
-      secretAccessKey: this.params.secretAccessKey,
-      s3ForcePathStyle: s3UrlBase.protocol === 'http:',
+    const s3UrlObj: any = new URL(this.url)
+    const regionRegex: RegExp = /^https?:\/\/s3\.([^.]+)\.amazonaws\.com/;
```

Comment:
`amazonaws` is not a valid sub-domain for test purposes, there should be `wasabisys` instead. Also, It should be configurable and not present in the code.

Reply:
OK, please fix PR.